### PR TITLE
Add ci/cd on PR for cypress e2e, and add developer read me

### PR DESCRIPTION
**Pull request by itailiors**
Implementing  (CI/CD) pipelines for Cypress end-to-end (E2E) testing.
This integration ensures that our E2E tests are automatically triggered upon a PR.

also adding a comprehensive README file tailored for developers/contributors.

### CODE REVIEW

**Comment by itailiors** [comment]
@dangershony @DavidGershony 
if possible the dev read me can be like this?
image

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
     "license": "MIT",
     "scripts": {
         "version": "node -p \"require('./package.json').version\"",
-        "changelog": "conventional-changelog -i CHANGELOG.md -s -r 0 && git add CHANGELOG.md"
+        "changelog": "conventional-changelog -i CHANGELOG.md -s -r 0 && git add CHANGELOG.md",
```
Comment (dangershony) [inline package.json:7]: What does this line do? does it generate and commit changelogs automatically?

**Review by dangershony — APPROVED** [review]

**Comment by dangershony** [comment]
Absolutely brilliant to have automated testing for ui

File: package.json (hunk 1 above)
Comment (itailiors) [reply package.json:7]: This line updates a changelog file and adds it to version control as far as i know,
was here before me i added the ", "😄

**Pull request merged**